### PR TITLE
exo: add auto-completion

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
- [x] bash

### CODE REVIEW

**Comment by greut** [comment]
no changes in `.travis` or `goreleaser`?

**Comment by pierre-emmanuelJ** [comment]
WIP, i will add after some tests

File: .travis.yml [hunk 1]
```diff
@@ -46,6 +46,8 @@ jobs:
       dep ensure -v -vendor-only
       cd ../..
       goreleaser --snapshot --skip-sign
+      cd cmd/exo
+      go run ./completion/main.go
```
Comment (greut) [inline .travis.yml]: the goal would be to include those files in the release, or do I miss something?

Comment (pierre-emmanuelJ) [reply .travis.yml]: yes

**Comment by greut** [comment]
for Bash it seems ok. Do you know why the zsh file contains so little?

```console
% exo 
affinitygroup  eip            privnet        sshkey         vm             
config         firewall       ssh            template       zone
```

**Comment by pierre-emmanuelJ** [comment]
i dont know but i think we will be forced to use bash_completion to make zsh completion like cs

File: .travis.yml [hunk 2]
```diff
@@ -45,6 +45,10 @@ jobs:
       cd ../exo
       dep ensure -v -vendor-only
```
Comment (greut) [inline .travis.yml]:
```
go run bash_completion/main.go
mv bash_completion ../..
```

File: .travis.yml [hunk 3]
```diff
@@ -45,6 +45,10 @@ jobs:
...
       cd ../..
+      cd cmd/exo/completion
+      go run main.go
+      cp bash_completion ../../../
```
Comment (greut) [inline .travis.yml]: or `go run cmd/exo/completion`

Comment (pierre-emmanuelJ) [reply .travis.yml]: 👍

File: .travis.yml (hunk 2 above)
Comment (pierre-emmanuelJ) [reply .travis.yml]: 👇

**Pull request merged**